src/main/java/it/pagopa/pn/deliverypush/action/startworkflow/notificationvalidation/NotificationValidationActionHandler.java

```diff
@@ -281,7 +281,7 @@ private void checkDocumentsMaxPageNumber(NotificationInt notification, int maxPa
                             safeStorageService.getFile(ref.getKey(), false)
                                     .onErrorResume(PnFileNotFoundException.class, this::handleNotFoundError))
                     .block();
-            byte[] pieceOfContent = safeStorageService.downloadPieceOfContent(fd.getKey(), fd.getDownload().getUrl(), -1).block();
+            byte[] pieceOfContent = safeStorageService.downloadPieceOfContent(Objects.requireNonNull(fd).getKey(), fd.getDownload().getUrl(), -1).block();
```

Comment:
questa vorrei capirla meglio, cosa accade se l'oggetto è null? Potrebbe avere senso essere più espliciti e mettere un if che copre tutto con il controllo in questione?

Reply:
Qui abbiamo implementato il suggerimento di intellij più "compatto" per risolvere la segnalazione. Analizzando un po' il contesto direi che quell'oggetto non dovrebbe essere mai null poichè viene restituito da una chiamata (getFile per filekey) a safe storage. Per quello che so io non mi sembra possibile avere in risposta un oggetto vuoto, al limite avremmo un'eccezione per file non trovato.